streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -230,10 +254,22 @@ synchronized void unlock(final TaskId taskId) throws IOException {
      *                       this amount of time (milliseconds)
      */
     public synchronized void cleanRemovedTasks(final long cleanupDelayMs) {
+        try {
+            cleanRemovedTasks(cleanupDelayMs, true);
+        } catch (final IOException cannotHappen) {
+            // dummy log to make findbugs happy
+            log.error("{} Failed to delete the state directory due to an unexpected exception.", logPrefix(), cannotHappen);
```

Comment:
Could we throw an IllegalStateException instead?

Reply:
We set `swallowException=true` -- wouldn't throwing an exception contradict this? Note, this code path is used for the task cleaner back ground thread -- an exception would kill this thread, what is the reason why I we want to swallow. Or do I miss anything?

Reply:
Throwing `IllegalStateException` is served as the purpose that "this should never happen, and if it does it is a bug and hence it is ok to fail and stop the world".

Reply:
Ok. I guess, we need to register the uncaught exception handler to the cleanup thread for this case, too.